src/ui/components/SearchInput/index.js

```diff
@@ -0,0 +1,69 @@
+/* global document, window */
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+import './style.scss';
+
+class SearchInput extends React.Component {
+  static propTypes = {
+    className: PropTypes.string,
+    inputRef: PropTypes.func,
+    name: PropTypes.string.isRequired,
+    placeholder: PropTypes.string,
+  }
+
+  componentDidMount() {
+    this.setIconPosition();
+    window.addEventListener('resize', this.setIconPosition);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.setIconPosition);
+  }
+
+  onBlur = () => {
+    if (!this.input.value) {
+      this.root.classList.remove('SearchInput--text');
```

Comment:
Are we using `--` anywhere else? It seems out of place, though I know there was a thought behind doing it.